clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -480,9 +481,6 @@ private long listOffset(TopicPartition partition, OffsetResetStrategy strategy)
     }
 
     private List<ConsumerRecord<K, V>> drainRecords(PartitionRecords<K, V> partitionRecords, int maxRecords) {
-        if (partitionRecords.isDrained())
-            return Collections.emptyList();
```

Comment:
That's right. I was trying to remove the need to handle empty fetch responses specially to update lag metrics. Before we simply discarded the fetch result if it had no records, but now we pass it through. Does this seem better or worse?